iml-agent/src/action_plugins/high_availability/mod.rs

```diff
@@ -216,17 +219,324 @@ pub async fn get_ha_resource_list(_: ()) -> Result<Vec<ResourceAgentInfo>, ImlAg
                 true
             }
         })
-        .map(|res| async move {
-            let output = crm_resource(&["--query-xml", "--resource", res]).await?;
-            if let Some(xml) = output.split("xml:").last() {
-                process_resource(xml.as_bytes())
-            } else {
-                Err(ImlAgentError::MarkerNotFound)
-            }
-        });
+        .map(str::to_string)
+        .collect())
+}
+
+pub async fn get_ha_resource_list(_: ()) -> Result<Vec<ResourceAgentInfo>, ImlAgentError> {
+    let xs = resource_list().await?.into_iter().map(|res| async move {
+        let output = crm_resource(&["--query-xml", "--resource", &res]).await?;
+        if let Some(xml) = output.split("xml:").last() {
+            process_resource(xml.as_bytes())
+        } else {
+            Err(ImlAgentError::MarkerNotFound)
+        }
+    });
     try_join_all(xs).await
 }
 
+fn xml_add_op(
+    res: &mut Element,
+    id: &str,
+    name: &str,
+    timeout: &Option<String>,
+    interval: &Option<String>,
+) {
+    if timeout.is_none() && interval.is_none() {
+        return;
+    }
+
+    let op = res.append_new_child("op");
+    op.set_attr("name", name);
+    if let Some(value) = interval {
+        op.set_attr("id", format!("{}-{}-timeout-{}", id, name, value))
+            .set_attr("interval", value);
+    }
+    if let Some(value) = timeout {
+        op.set_attr("id", format!("{}-{}-timeout-{}", id, name, value))
```

Comment:
Is it possible both timeout and interval are Some? What would be the id if `op`?
Why isn't `op` mutable? :)

Reply:
Possible: yes (Monitor operation can techincally have both an interval and a timeout, but practically, not really).
If both set, it would be "interval", but id doesn't really matter, it just needs to be unique.
Compiler says it doesn't need to be mutable.